lib/TopStoriesFeed.jsm

```diff
@@ -448,18 +477,27 @@ this.TopStoriesFeed = class TopStoriesFeed {
     this.init();
   }
 
-  onAction(action) {
+  async onAction(action) {
+    let updateProps = {};
     switch (action.type) {
       case at.INIT:
         this.init();
         break;
       case at.SYSTEM_TICK:
         if (Date.now() - this.storiesLastUpdated >= STORIES_UPDATE_TIME) {
-          this.fetchStories();
+          await this.fetchStories();
         }
         if (Date.now() - this.topicsLastUpdated >= TOPICS_UPDATE_TIME) {
-          this.fetchTopics();
+          await this.fetchTopics();
+        }
+
+        if (this.stories) {
+          updateProps = {rows: this.stories};
+        }
+        if (this.topics) {
+          updateProps = {...updateProps, topics: this.topics, read_more_endpoint: this.read_more_endpoint};
         }
+        this.dispatchUpdateEvent(false, updateProps);
```

Comment:
Is there anyway you could create a function to handle both the SYSTEM_TICK and INIT updates in one spot?